src/js/components/CombinationMenuItem.vue

```diff
@@ -61,7 +64,7 @@ export default {
   methods: {
     fetchChildren() {
       // we'll lazy load children, but cache them
-      if (this.lastFetch !== this.requestType[this.menuType]) {
+      if (this.children[this.menuType].length < 1) {
```

Comment:
Maybe we can think how to do this not as `this.children.subType` but rather `this.texts` and `this.images`? Or something like that, to avoid the nested object in the data. It seems to me like we start getting into weird situations with Vue's reactivity

Reply:
You would know more about the reactivity with nested objects than me.  But is this not an acceptable data structure, especially in regard to code clarity?  We can have children for this node, and those children can be of two types (we might even add more in the future), so children has its own JSON object for each type of subnode.  If we use two (three, etc...) independent variables for each node type, it seems we decrease visibility that those variables are used to generate the subcomponents (though we could use variable names like textChildren and imageChildren).  Let me know what your thoughts are, especially in regard to reactivity in nested data.